app/src/main/java/com/zulip/android/activities/MessageListFragment.java

```diff
@@ -131,6 +131,7 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         View view = inflater.inflate(R.layout.fragment_message_list, container,
                 false);
 
+        messageIndex = new SparseArray<>();
```

Comment:
unfortunately, this didn't quite work for me in case of pm notifications; after narrowing to pm (using show action) I couldn't see the latest message added to the list and on pressing back the message list doubled (old messages already loaded in the list were added again at the bottom).